RecommenderSystems/dcn/README.md

```diff
@@ -0,0 +1,167 @@
+# Deep&Cross
+ [Deep & Cross Network](https://dl.acm.org/doi/10.1145/3124749.3124754) (DCN) can not only keep the advantages of DNN model, but also learn specific bounded feature crossover more effectively. In particular, DCN can explicitly learn cross features for each layer without the need for manual feature engineering, and the increased algorithm complexity is almost negligible compared with DNN model.
+ ![DCN](https://user-images.githubusercontent.com/80230303/159417248-1975736f-3de8-4972-84e3-2f0f346cbc1a.png)
+
+
+Oneflow API is compatible to Pytorch, so only minor modification in codes then we can apply the Pytorch implemented modules to Oneflow. Therefore, We refer to some DCN implementation in [FuxiCTR](https://github.com/xue-pai/FuxiCTR/tree/v1.0.2) such as CrossNet. We follow the FuxiCTR [CrossNet implementation](https://github.com/xue-pai/FuxiCTR/blob/0f4ccf36aac02d120cb0c87f417c40e0e4330744/fuxictr/pytorch/models/DCN.py#L77), reuse the `CrossInteractionLayer` as a basic Crossnet layer, and make it multi layers in `CrossNet`. The only difference of CrossNet implementation between FuxiCTR and Oneflow is `nn` module, one is `torch.nn` and another one is `oneflow.nn` .
```

Comment:
Therefore, we adopted some implementation from [FuxiCTR](https://github.com/xue-pai/FuxiCTR/tree/v1.0.2), for example the `CrossInteractionLayer` was reused as a basic `CrossNet` hidden layer, and we make a recurrent loop of these hidden layers in `CrossNet` module to get high-degree interaction across features.

~The only difference of CrossNet implementation between FuxiCTR and Oneflow is `nn` module, one is `torch.nn` and another one is `oneflow.nn` .~